src/main/java/org/opentripplanner/transit/model/framework/TransitEntity.java

```diff
@@ -18,8 +21,12 @@
  *     frequently use this to index and lookup entities.
  *   </li>
  * </ol>
+ * This class also enforce a strong type-safe relationship between entity and builder.
  */
-public abstract class TransitEntity implements Serializable {
+public abstract class TransitEntity<
```

Comment:
@hannesj would you mind changing the hasCode to:
```
@Override
  public final int hashCode() {
    return id.hashCode();
  }
```
The change is maybe small, but
 - it obstruct reading
 - prevent fancy tricks overriding the get methods...

Reply:
Done